Synching the repository with latest API changes

Ayodhya94 opened the pull request:
## Purpose
Fixes #34

## Goals
> Synching the repository with `ballerina-lang` master branch.

## Approach
> Changing the code to match the latest API changes.

dulajdilshan commented:
Don't we `module-info.java` file with this?

Ayodhya94 commented:
> Don't we need `module-info.java` file with this?

With it, I always got `not found module` error. Once I removed it worked fine. Therefore I removed that file.

dulajdilshan commented on gradle.properties:
```diff
@@ -3,5 +3,6 @@ org.gradle.jvmargs='-Dfile.encoding=UTF-8'
-ballerinaLangVersion=2.0.0-Preview5-SNAPSHOT
+ballerinaLangVersion=2.0.0-Preview6-SNAPSHOT
 stdlibIoVersion=0.5.0
+slf4jVersion=1.7.30
```
Do we need `slf4jVersion` here?

Ayodhya94 replied:
This is removed

pubudu91 commented on spotbugs-exclude.xml:
```diff
@@ -0,0 +1,9 @@
+<FindBugsFilter>
+    <Match>
+    <Bug pattern="URF_UNREAD_FIELD"/>
+    <Bug pattern="URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD"/>
```
Do we need to skip these? Doesn't these mean that there are unused fields in the code? If that's the case, shall we remove them, instead of adding a exclusions?

Ayodhya94 replied:
There's a single field that causes this error in spotbugs. But we cannot remove that as it causes a test failure.

BuddhiWathsala commented:
> > Don't we need `module-info.java` file with this?
> 
> With it, I always got `not found module` error. Once I removed it worked fine. Therefore I removed that file.

I guess we might need this `module-info.java` when it comes to exporting classes. Please check data mapper use cases with the final ballerina pack and check whether the data mapper use cases receive class not found issues?.

BuddhiWathsala commented:
The build is failing because you send this PR from a forked branch. In order to solve this, you can follow this PR (https://github.com/ballerina-platform/module-ballerina-io/pull/34/files). The build failure occurred when it couldn't resolve IO stdlib. Anyhow, I didn't find any usage of the IO module even though it imported.

Ayodhya94 commented:
> The build is failing because you send this PR from a forked branch. In order to solve this, you can follow this PR (https://github.com/ballerina-platform/module-ballerina-io/pull/34/files). The build failure occurred when it couldn't resolve IO stdlib. Anyhow, I didn't find any usage of the IO module even though it imported.


Thank you. Added the suggested edits.

Ayodhya94 commented:
> > > Don't we need `module-info.java` file with this?
> > 
> > 
> > With it, I always got `not found module` error. Once I removed it worked fine. Therefore I removed that file.
> 
> I guess we might need this `module-info.java` when it comes to exporting classes. Please check data mapper use cases with the final ballerina pack and check whether the data mapper use cases receive class not found issues?.

I tried building a few ballerina connectors, where this module is used to extract data. I got build errors but non were due to the data mapper. I had a talk with Miyuru ayya as well and think it's safe to merge now.
